mqtt/client.go

```diff
@@ -114,12 +114,28 @@ func NewClient(ctx log.Interface, id, username, password string, brokers ...stri
 	}
 }
 
-func (c *defaultClient) Connect() error {
+var (
+	// ConnectRetries says how many times the client should retry a failed connection
+	ConnectRetries = 5
+	// ConnectRetryDelay says how long the client should wait between retries
+	ConnectRetryDelay = time.Second
+)
+
+func (c *defaultClient) Connect() (err error) {
 	if c.mqtt.IsConnected() {
 		return nil
 	}
-	if token := c.mqtt.Connect(); token.Wait() && token.Error() != nil {
-		return fmt.Errorf("Could not connect: %s", token.Error())
+	for retries := 0; retries < ConnectRetries; retries++ {
+		token := c.mqtt.Connect()
+		token.Wait()
+		err = token.Error()
+		if err == nil {
+			break
+		}
+		<-time.After(time.Duration(retries) * ConnectRetryDelay)
```

Comment:
`ConnectRetryDelay`'s comment indicates that it is the time to wait between retries, but here you multiply it by the number of retries so far. Either update command or get rid of multiplication. Also, on the first retry, `retries` is 0 so there's no wait